clients/drcachesim/tracer/instru.h

```diff
@@ -238,10 +236,12 @@ class instru_t {
     virtual int
     instrument_memref(void *drcontext, void *bb_field, instrlist_t *ilist, instr_t *where,
                       reg_id_t reg_ptr, int adjust, instr_t *app, opnd_t ref,
-                      int ref_index, bool write, dr_pred_type_t pred) = 0;
+                      int ref_index, bool write, dr_pred_type_t pred,
+                      bool memref_needs_full_info) = 0;
     virtual int
     instrument_instr(void *drcontext, void *tag, void *bb_field, instrlist_t *ilist,
-                     instr_t *where, reg_id_t reg_ptr, int adjust, instr_t *app) = 0;
+                     instr_t *where, reg_id_t reg_ptr, int adjust, instr_t *app,
+                     bool memref_needs_full_info, uintptr_t mode) = 0;
```

Comment:
We want instru_t & subclasses to be as modular and separated from the tracer details as we can get them so at first I thought this was exposing the particular bbdup modes which we do not want to do.  But I see in the code this is a pass-through.  Can we either A) delete this parameter here and obtain the mode via a new load-acquire in the callback (I don't think that should case any overhead concern as it is not on a fastpath), which I would vote for; or B) document that this is an opaque value just passed through to the insert_load_buf callback.

Reply:
This is using the mode value that is passed to event_app_instruction_case and not the current tracing_mode value. So I think A is not an option. I will document this.